_inc/client/components/multiple-choice-question/index.jsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { memoize, pick, shuffle, values } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import MultipleChoiceAnswer from './answer';
+import { FormFieldset } from 'components/forms';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const shuffleAnswers = memoize(
+	answers => {
+		const shuffles = shuffle( answers.filter( ( { doNotShuffle } ) => ! doNotShuffle ) );
+		return answers.map( answer => ( answer.doNotShuffle ? answer : shuffles.pop() ) );
```

Comment:
`return answers.map( answer => ( answer.doNotShuffle ? answer : shuffles.pop() ) );`

There's part of me that doesn't want to see something stateful in a call to `map`, and another part of me that thinks this is brilliant. I like it :)

Reply:
heh, I will take "brilliant